src/email/email.js

```diff
@@ -0,0 +1,120 @@
+const { SMTPServer } = require("smtp-server");
+const { MailParser } = require('mailparser');
+const ParseEmailAddress = require("email-addresses");
+const { Logging } = require('../log');
+
+const log = Logging.get("email");
+
+const roomAlias = function(rcptTo, mxDomain) {
+    let localpartRcptTo;
+    for (let i = 0; i < rcptTo.length; i++) {
+        localpartRcptTo = ParseEmailAddress.parseOneAddress(rcptTo[i].address).local;
+        if ( localpartRcptTo.endsWith(mxDomain)) {
+            log.info("Message destination address:", localpartRcptTo);
+            break;
+        }
+    }
+    return getUserIdorAlias(localpartRcptTo);
+};
+const splitAt = index => x => [x.slice(0, index), x.slice(index+1)];
+
+const getUserIdorAlias = function(localPart) {
+
+    // Received room+<roomalias>_hs
+    if (localPart.startsWith('room+')) {
+        log.info("Message destination is a room");
+        let alias = localPart.substring(localPart.indexOf('+')+1);
+        if (alias.lastIndexOf('_') >= 1) {
+            let res = splitAt(alias.lastIndexOf('_'))(alias);
+            res.unshift("room");
+            return res;
+        }
+      return null;
+    }
+
+    // Received user+<username>_<hs>
+    else if (localPart.startsWith('user+')) {
+        log.info("Message destination is a user");
+        let uname = localPart.substring(localPart.indexOf('+')+1);
+        if (uname.lastIndexOf('_') >= 1) {
+            return splitAt(uname.lastIndexOf('_'))(uname).unshift("user");
+        }
+      return null;
+    }
+
+    log.warn("Destination is not valid");
+    return null;
+};
```

Comment:
So the choice is yours here, but I tend to throw on failures with an error message rather than logging and returning null. The caller should then handle this failure. I'll leave it up to you as a taste thing.